tlatools/org.lamport.tlatools/test/tlc2/util/BufferedRandomAccessFileTest.java

```diff
@@ -170,4 +168,155 @@ public void testInvalidateBufferedData() throws IOException {
+		try (final java.io.RandomAccessFile raf = new BufferedRandomAccessFile(tmpFile, "rw")) {
+			raf.write(83);
+			raf.seek(0);
+			byte[] buffer = new byte[2];
+			int n = raf.read(buffer, 0, 2);
+			Assert.assertEquals(1, n);
+			Assert.assertEquals(83, buffer[0]);
+		}
+	}
+
+	@Test
+	public void regressionTest05() throws IOException {
+		final File tmpFile = File.createTempFile("tmp", ".bin");
+		tmpFile.deleteOnExit();
+		try (final java.io.RandomAccessFile raf = new BufferedRandomAccessFile(tmpFile, "rw")) {
+			byte[] b1 = { 13, 41, -113, 61, -7 };
+			raf.write(b1, 3, 1);
+			raf.write(117);
+			byte[] b2 = { 63, -59, 84, -4, 7 };
+			raf.write(b2, 1, 1);
+			raf.seek(2);
+
+			// REMINDER: read() returns an UNSIGNED byte in the range [0, 255].  So even
+			// though 197 does not appear in our writes, it is still the correct output
+			// since it is the unsigned representation of the byte -59.
+			Assert.assertEquals(197, raf.read());
+		}
+	}
+
+	@Test
+	public void regressionTest06() throws IOException {
+		final File tmpFile = File.createTempFile("tmp", ".bin");
+		tmpFile.deleteOnExit();
+		try (final java.io.RandomAccessFile raf = new BufferedRandomAccessFile(tmpFile, "rw")) {
+			raf.seek(9037);
+			raf.setLength(8311);
+			byte[] b1 = { 114, -128, 118, -15, -71, -43, -70, -27, 122, -33, -8, 80, -75, -7, -123, -88, -109, 4, 124, -69, 73, -86, 21, -63, 84, -13, 106, 81, 64, 69, -109, -56, -87, -48, 66, -98, -83, 44, 111, -77, 78, 17, -123, 115, -18, -55, -42, -75, 73, -73, -23, -125, -113, -39, 85, -39, -119, -16, 76, -106, -47, -72, 113, 25, 78, -90, 35, -2, -109, 1, 63, 43, 14, 44, -112, 8, -15, 15, 18, -82, -20, 13, 63, 86, -35, -92, -71, 85, 1, 82, 116, 29, 112, -53, -31, 10, 5, 37, 84, 96, -43, 2, -11, 10, 23, 50, -100, 72, -17, -124, -43, 14, -69, -98, 34, 119, 46, -30, -71, 45, 69, 31, -81, 60, 48, 37, 59, 15, 69, -91, -100, 125, 35, -43, 97, 106, -71, 24, -90, 26, -4, -117, 79, -89, 6, 46, -90, 52, 3, -2, 71, 121, 13, 14, 14, -82, 71, 82, -84, -31, -37, -6, 108, -84, -16, 23, -29, -125, -70, -78, -83, -23, -111, 10, -51, -51, -76, -36, 46, -70, -30, -57, 112, 34, -54, -108, -16, -5, -114, -93, -3, 109, 101, 11, -119, -102, -14, 121, 22, -112, -47, 106, 96, -6, 8, -47, -7, 42, -66, -90, -60, -50, -43, 78, -36, -10, 29, -51, -112, 26, 4, -97, -97, -84, -40, -76, 65, -82, -91, -35, 42, -56, 44, -50, 112, -12, 36, -109, -107, -120, -79, 49, 99, -123, -25, -51, -68, -79, 94, 38, 57, 81, 75, 0, -25, -70, -5, 3, 116, -74, -55, -115, -127, -34, -21, -26, 100, 122, -122, 112, -97, 71, 48, -21, 37, -128, -85, -88, -2, 86, 22, -24, -115, -16, 41, 73, -104, -47, -22, -70, -56, 125, -32, -127, -111, -63, -94, 102, -27, 121, 56, 13, 58, -32, 112, 119, 105, -24, -75, 119, -53, -64, -123, -124, -108, -42, 9, -30, 36, 30, -109, 39, -18, 42, -79, -78, 62, 73, -37, -96, -41, 85, 39, 58, -54, 19, -69, 111, 60, -89, 30, 37, -78, -112, 29, -69, 98, 108, 120, -54, 59, -19, 60, 74, -58, 20, -56, 126, 23, -125, -110, 85, -52, 42 };
```

Comment:
nit: this seems a bit overkill.  Can't this just be a `raf.write(0)` and then assert that length is 8312?